k8s/plugin/src/resources/upgrade.rs

```diff
@@ -54,57 +61,101 @@ impl UpgradeResources {
         })
     }
 
-    /// Install the upgrade resources
-    pub async fn install(ns: &str) {
-        match UpgradeResources::new(ns).await {
-            Ok(uo) => {
-                let pp = PostParams::default();
-                // Create a service account
-                match uo.get_service_account().await {
-                    Ok(sa) => {
-                        if sa.iter().count() == 0 {
+    /// Create/Delete service action
+    pub async fn service_account_actions(&self, ns: &str, action: Actions) -> Result<(), ()> {
+        match self.get_service_account().await {
+            Ok(sa) => {
+                if sa.iter().count() == 0 {
+                    match action {
+                        Actions::Create => {
                             let ns = Some(ns.to_string());
                             let service_account = objects::upgrade_operator_service_account(
                                 ns,
-                                uo.release_name.clone(),
+                                self.release_name.clone(),
                             );
                             let pp = PostParams::default();
-                            match uo.service_account.create(&pp, &service_account).await {
+                            match self.service_account.create(&pp, &service_account).await {
```

Comment:
what's the difference between sa (line 67), and service_account (72) ?
I'm trying to see if we can do the match action at the root and not do the sa.iter.count.

Reply:
Line 67 `sa` is the list of Service account.   Basically this ` pub async fn get_service_account(&self) -> Result<ObjectList<ServiceAccount>, Error>` . I am doing an `sa.iter().count()` as its return an object list . We can return `ServiceAccount` only instead of the list, but that IMHO wont be good as we will be restricting to just a ServiceAccount at index 0. Returning the list is good as this code can be reused whenever we need to have a list of service accounts.

Reply:
I get that, what I'm asking what is the different between those two variables, `sa` and `service_account`?

Reply:
changes the code where in i am using `get_apt` rather than list.